packages/atomic/src/components/search/tabs/atomic-tab-section/atomic-tab-section.tsx

```diff
@@ -0,0 +1,185 @@
+import {InitializeBindings} from '../../../../utils/initialization-utils';
+import {getActiveTab} from '../../../../utils/tab-utils';
+import {Button} from '../../../common/button';
+import {Bindings} from '../../atomic-search-interface/atomic-search-interface';
+
+/**
+ * @internal
+ */
+@Component({
+  tag: 'atomic-tab-section',
+  styleUrl: 'atomic-tab-section.pcss',
+  shadow: true,
+})
+export class AtomicTabSection {
+  @InitializeBindings() public bindings!: Bindings;
+  @Element()
+  private host!: HTMLElement;
+
+  /**
+   * When provided, this tab will be selected by default when the component loads. Otherwise, the first tab is selected automatically.
+   */
+  @Prop() defaultActiveTab?: string;
+  /**
+   * Specifies whether the state should be cleared when the tab changes.
+   */
+  @Prop() clearStateOnTabChange?: boolean = false;
+
+  @State() public error!: Error;
+  @State() tabs: HTMLAtomicTabElement[] = [];
+
+  @Event({
+    eventName: 'atomic/tabInit',
+  })
+  tabInit!: EventEmitter;
+
+  @Listen('atomic/tabClick')
+  handleTabClick() {
+    this.updateActiveTab();
+    if (this.clearStateOnTabChange) {
+      this.clearState();
+    }
+  }
+
+  componentDidLoad() {
+    this.setInitialTab();
+  }
+
+  componentWillUpdate() {
+    const tabs = [...this.host.querySelectorAll('atomic-tab')];
+    this.tabs = tabs;
+
+    if (tabs.length === 0) {
+      this.error = new Error(
+        'The "atomic-tab-section" element requires at least one "atomic-tab" child.'
+      );
+      return;
+    }
+  }
+
+  setInitialTab() {
+    const initialTab =
+      this.tabs.find((tab) => tab.name === this.defaultActiveTab) ||
+      this.tabs[0];
+    const activeTab = getActiveTab(this.bindings.engine.state)?.tab;
+
+    if (initialTab && !activeTab) {
+      this.bindings.engine.dispatch(
+        loadTabSetActions(this.bindings.engine).updateActiveTab(initialTab.name)
+      );
+      this.tabInit.emit();
+    }
+    this.updateActiveTab();
+  }
+
+  buildDropdown() {
+    return this.tabs.map((tab) => (
+      <li>
+        <Button
+          style="text-transparent"
+          title={tab.name}
+          class="w-full px-6 pb-1 text-xl text-neutral-dark dropdown-option"
+          text={tab.label}
+          onClick={() => {
+            tab.select();
+          }}
+        />
+      </li>
+    ));
+  }
+
+  updateActiveTab() {
+    const dropdownButton =
+      this.host.shadowRoot?.querySelector('.dropdown-button');
+    const dropdownOptions = [
+      ...(this.host.shadowRoot?.querySelectorAll('.dropdown-option') || []),
+    ];
+    const activeTabName = getActiveTab(this.bindings.engine.state)?.tab;
+
+    this.tabs.forEach((tab) => {
+      tab.isActive = tab.name === activeTabName;
+
+      if (tab.isActive) {
+        const span = dropdownButton?.getElementsByTagName('span')[0];
+        if (span) {
+          span.textContent = tab.label;
+        }
+      }
+    });
+
+    dropdownOptions.forEach((option) => {
```

Comment:
As explained in my earlier comment, all of this should be done in the render function `buildDropdown` to check if the current button being rendered is the active one in Headless state